src/PhotosphereViewer.tsx

```diff
@@ -205,8 +216,16 @@ function PhotosphereViewer({
       // setCurrentPhotosphere has to be used to get the current state value because
       // the value of currentPhotosphere does not get updated in an event listener
       setCurrentPhotosphere((currentState) => {
-        const passMarker = currentState.hotspots[marker.config.id];
-        setHotspotArray([passMarker]);
+        //const passMarker = currentState.hotspots[marker.config.id];
+        setHotspotArray((prevArray) => {
+          const passMarker = currentState.hotspots[marker.config.id];
+          return [...prevArray, passMarker];
+        }); 
```

Comment:
What's the purpose of this change?

`hotspotArray` represents the stack of (nested) hotspots, so it shouldn't be storing unrelated hotspots.

Is it even possible for `prevArray` to not be `[]` here? I thought the only way to reach this code is by clicking on a hotspot while the popover is closed.